docs/develop/go/versioning.mdx

```diff
@@ -143,40 +180,62 @@ if v == workflow.DefaultVersion {
+Since Workflow Executions that were started before `GetVersion` was added to the code won't have the associated Marker in their Event History, you'll need to use a different query to determine if any of those are still running:
 
-- workflow.ExecuteActivity()
-- workflow.ExecuteChildWorkflow()
-- workflow.NewTimer()
-- workflow.RequestCancelWorkflow()
-- workflow.SideEffect()
-- workflow.SignalExternalWorkflow()
-- workflow.Sleep()
+```
+WorkflowType = "PizzaWorkflow" 
+    AND ExecutionStatus = "Running" 
+    AND TemporalChangeVersion IS NULL
+```
 
-Adding, removing, or reordering any of the preceding methods triggers the sanity check and results in a nondeterminism error.
+If you have found that there are no longer any open executions for the first two versions of the Workflow, for example, then you could remove support for them by changing the code as shown below:
 
-The sanity check does not perform a thorough check.
-For example, it does not check on the Activity's input arguments or the Timer duration.
-If the check is enforced on every property, it becomes too restrictive and harder to maintain the Workflow code.
-For example, if you move your Activity code from one package to another package, that move changes the `ActivityType`, which technically becomes a different Activity.
-But we don't want to fail on that change, so we check only the function name part of the `ActivityType`.
+```go
+version := GetVersion(ctx, "ChangedNotificationActivityType", 2, 3)
+if version  == 2 {
+    err = workflow.ExecuteActivity(ctx, SendTextMessage).Get(ctx, nil)
+} else {
+    err = workflow.ExecuteActivity(ctx, SendTweet).Get(ctx, nil)
+}
+```
 
-## How to use Worker Versioning in Go {#worker-versioning}
+When `workflow.GetVersion()` is run for a new Workflow execution, it records a marker in the Workflow history so that all future calls to `GetVersion()` for this change ID — Step 1 in the example — on this Workflow execution will always return the given version number.
 
-Documentation for this feature is coming soon!
+Patching allows you to make changes to currently running Workflows.
+It is a powerful method for introducing compatible changes without introducing non-determinism errors.
 
-For now, please join #safe-deploys in our [Community Slack](https://temporal.io/slack) to find the latest status and pre-release docs.
+## Worker Versioning
 
-If you were using a previous pre-release version of Worker Versioning, it's now deprecated.
+Temporal's [Worker Versioning](/production-deployment/worker-deployments/worker-versioning) feature allows you to tag your Workers and programmatically roll them out in deployment versions, so that old Workers can run old code paths and new Workers can run new code paths. This way, you can pin your deployments to specific revisions, avoiding the need for patching.
 
-See [Legacy Worker Versioning](worker-versioning-legacy) if you still need those docs.
+## Runtime checking {#runtime-checking}
```

Comment:
This section I think would be better as a link out to what I'm adding here: https://github.com/temporalio/documentation/pull/3614

Which has a more complete list, along with more context and allows us to avoid repeating it per-language